code/modules/power/apc.dm

```diff
@@ -749,7 +749,7 @@
 
 
 /obj/machinery/power/apc/proc/get_malf_status(mob/living/silicon/ai/malf)
-	if(istype(malf) && malf.mind.has_antag_datum(/datum/antagonist/traitor) || malf.malf_picker) // If they're a traitor OR they have the malf picker from the combat module
+	if(istype(malf) && (malf.mind.has_antag_datum(/datum/antagonist/traitor) || malf.malf_picker)) // If they're a traitor OR they have the malf picker from the combat module

```

Comment:
Early return is recommended both to eliminate excessive nesting, and to make stuff easier to read and prevent this kind of mistake.
I.e.
```js
if(!istype(malf))
  return 0
if(!malf.mind.has_antag_datum(/datum/antagonist/traitor))
  reuturn 0
if(!malf.malf_picker)
  return 0
```

Reply:
The given solution is incorrect. But I get what you mean. Will change once able